src/components/form/Form.js

```diff
@@ -478,13 +478,11 @@ export default class FormComponent extends Component {
 
   /**
    * Submit the form before the next page is triggered.
+   * @param {boolean} next
    */
   beforePage(next) {
     // Should not submit child forms if we are going to the previous page
-    if (!next) {
-      return super.beforePage(next);
-    }
-    return this.submitSubForm(true).then(() => super.beforePage(next));
```

Comment:
Travis, I agree, in that case we can left it as it is, because there is a check on the server which prevents resubmitting child submission if it is a POST request and child submission already has an id